### ISSUE TITLE
[Search] Scrolling Results

### ISSUE THREAD

**Issue by GammaGames** [fomantic/Fomantic-UI#771]
# Feature Request

## Description
I think it would be nice for the option to have a scrollable results menu. Sometimes users want a lot of results, and limiting them just makes them frustrated because they can't find what they (vaguely) searched for.

## Example
```css
.ui.search .results {
  max-height: 30vh;
  overflow-y: auto;
}
```

We could even reuse [variables](https://github.com/fomantic/Fomantic-UI/blob/master/src/definitions/modules/dropdown.less#L1126) from the [scrolling dropdown](https://semantic-ui.com/modules/dropdown.html#scrolling) options.

One issue with this is we would need to copy the dropdown's scrolling behavior when using keyboard shortcuts.

## Testcase
https://jsfiddle.net/nstquf0b/

## Screenshot
![image](https://user-images.githubusercontent.com/7832163/58360139-44e14280-7e44-11e9-80fe-799aefa14a7d.png)

Related to Semantic-Org/Semantic-UI/issues/3389, but I didn't understand how Jack's answer is a solution with the current implementation.

**Comment by lubber-de** [comment]
Basically a good and reasonable idea. :+1: 
I suggest to use the same approach as we already have for `dropdown`, because a fixed setting of 30vh could be less than wanted (especially on mobile screens) and it would also be a breaking change.
To avoid a breaking change (user currently can define the number of displayed entries (without a scrollbar) via `maxResults` ) i suggest to implement a `scrolling` class, just like dropdown, which implements the overflow-y scrollbar and a max height calculated from a predefined number of entries per screen size. If `scrolling` is set, the code should also not splice the resultset to the `maxResult` entries anymore but render all of the results instead (because a scrollbar appears now)

To let the user adjust the height according to their needs and viewport height we should add `long`, `very long`,`short` and `very long` as we already have for `dropdown`.

I would adopt the same logic from dropdown https://github.com/fomantic/Fomantic-UI/blob/b1327153b6f227cd0abd3cec7f0168db150ab537/src/themes/default/modules/dropdown.variables#L177-L190
and
https://github.com/fomantic/Fomantic-UI/blob/b1327153b6f227cd0abd3cec7f0168db150ab537/src/definitions/modules/dropdown.less#L438-L454

**Comment by lubber-de** [comment]
@GammaGames I should have read your post more accuratly...you already pointed to the dropdown variable setting... 🙄 😄

**Comment by GammaGames** [comment]
> a fixed setting of 30vh could be less than wanted (especially on mobile screens)

Yeah, I normally just use 50-30vh for my own projects because they're only expected to be used on desktop. I've done it so many times I thought I'd suggest it though

**Comment by exoego** [comment]
FYI: Fomantic recently released columnar variation of dropdown, which could be useful for showing bigger search results.
https://fomantic-ui.com/modules/dropdown.html#columnar-menu

The suggested enhancement for scrolling is good to have, of course.

**Comment by lubber-de** [comment]
Implemented by #1350
